orchagent/fdborch.cpp

```diff
@@ -491,168 +684,389 @@ void FdbOrch::doTask(NotificationConsumer& consumer)
     }
 }
 
-void FdbOrch::flushFDBEntries(sai_object_id_t bridge_port_oid,
-                              sai_object_id_t vlan_oid)
-{
-    vector<sai_attribute_t>    attrs;
-    sai_attribute_t            attr;
-    sai_status_t               rv = SAI_STATUS_SUCCESS;
-
-    SWSS_LOG_ENTER();
-
-    if (SAI_NULL_OBJECT_ID == bridge_port_oid &&
-        SAI_NULL_OBJECT_ID == vlan_oid)
-    {
-        SWSS_LOG_WARN("Couldn't flush FDB. Bridge port OID: 0x%" PRIx64 " bvid:%" PRIx64 ",",
-                      bridge_port_oid, vlan_oid);
-        return;
-    }
-
-    if (SAI_NULL_OBJECT_ID != bridge_port_oid)
-    {
-        attr.id = SAI_FDB_FLUSH_ATTR_BRIDGE_PORT_ID;
-        attr.value.oid = bridge_port_oid;
-        attrs.push_back(attr);
-    }
-
-    if (SAI_NULL_OBJECT_ID != vlan_oid)
-    {
-        attr.id = SAI_FDB_FLUSH_ATTR_BV_ID;
-        attr.value.oid = vlan_oid;
-        attrs.push_back(attr);
-    }
-
-    SWSS_LOG_INFO("Flushing FDB bridge_port_oid: 0x%" PRIx64 ", and bvid_oid:0x%" PRIx64 ".", bridge_port_oid, vlan_oid);
-
-    rv = sai_fdb_api->flush_fdb_entries(gSwitchId, (uint32_t)attrs.size(), attrs.data());
-    if (SAI_STATUS_SUCCESS != rv)
-    {
-        SWSS_LOG_ERROR("Flushing FDB failed. rv:%d", rv);
-    }
-}
-
 void FdbOrch::updatePortOperState(const PortOperStateUpdate& update)
 {
     SWSS_LOG_ENTER();
     if (update.operStatus == SAI_PORT_OPER_STATUS_DOWN)
     {
         swss::Port p = update.port;
-        flushFDBEntries(p.m_bridge_port_id, SAI_NULL_OBJECT_ID);
+    	flushFdbByPort(p.m_alias, 0);
     }
     return;
 }
 
 void FdbOrch::updateVlanMember(const VlanMemberUpdate& update)
 {
+    Port port;
+
+    string port_name = update.member.m_alias;
+    string vlan_name = update.vlan.m_alias;
+
     SWSS_LOG_ENTER();
 
+    if (!m_portsOrch->getPort(port_name, port))
+    {
+        SWSS_LOG_ERROR("could not locate port from alias %s", port_name.c_str());
+        return;
+    }
+
     if (!update.add)
     {
-        swss::Port vlan = update.vlan;
-        swss::Port port = update.member;
-        flushFDBEntries(port.m_bridge_port_id, vlan.m_vlan_info.vlan_oid);
+        if(port.m_type != Port::TUNNEL)
+        {
+            flushFdbByPortVlan(port_name, vlan_name, 1);
+        }
         return;
     }
 
-    string port_name = update.member.m_alias;
     auto fdb_list = std::move(saved_fdb_entries[port_name]);
     saved_fdb_entries[port_name].clear();
     for (const auto& fdb: fdb_list)
     {
-        // try to insert an FDB entry. If the FDB entry is not ready to be inserted yet,
-        // it would be added back to the saved_fdb_entries structure by addFDBEntry()
-        (void)addFdbEntry(fdb.entry, fdb.type);
+        for (const auto& fdb: fdb_list)
```

Comment:
The additional inner loop looks like a mistake. Did it ever compiled? I got an error on gcc 8.3. Probably you want a single-level loop.